Makefile.Common

```diff
@@ -12,10 +12,18 @@ ALL_SRC_AND_DOC := $(shell find . \( -name "*.md" -o -name "*.go" -o -name "*.ya
 # ALL_PKGS is used with 'go cover'
 ALL_PKGS := $(shell go list $(sort $(dir $(ALL_SRC))) 2>/dev/null)
 
-GOTEST_OPT?= -race -timeout 30s
+# build tags required by any component should be defined as an independent variables and later added to GO_BUILD_TAGS below
+#
+# podmanreceiver uses podman Go bindings which is a bit coupled with core podman and as a result
+# requires the use of CGO and presence of some C libraries. The following build tags either turn off
+# these features or enforce using native Go alternatives. 
+# References: https://podman.io/getting-started/installation#build-tags and https://github.com/containers/podman/issues/6866
+GO_PODMAN_BUILD_TAGS=containers_image_openpgp,exclude_graphdriver_btrfs,exclude_graphdriver_devicemapper
```

Comment:
Can we move these exception to the podman's makefile? It is a bit untidy that the common makefile has anything about a specific component.

Reply:
that will only solve running tests for podmanreceiver but we need these for the main build as well.

Reply:
Can we come with a way for components to inject exceptions like this into the build? Perhaps an include file that each component may have which the main build uses?
I would not want to pollute the main makefile with component-specific rules or code like this.
Maybe you can come with some nice approach in a separate PR after merging this one?

Reply:
Sure, I can do that but not sure if the added build complexity will be worth it given we only have a single component that needs this today. I'll create an issue for it and assign to myself once this is merged.

Reply:
I think this is also something to consider when it comes to different distros using this receiver and https://github.com/open-telemetry/opentelemetry-collector-builder in general. This should at the very least be documented within the receiver folder and ideally (for future work) we would have a system for each component to declare its build options